### PR TITLE
Remove path-to-regexp and add koa-route definitions

### PR DESCRIPTION
**Pull request by migstopheles**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If removing a declaration:
- [x] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/koa-route/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    search: RouteMethod;
+    subscribe: RouteMethod;
+    trace: RouteMethod;
+    unbind: RouteMethod;
+    unlink: RouteMethod;
+    unlock: RouteMethod;
+    unsubscribe: RouteMethod;
+}
+
+declare const routes: KoaRoutes;
```
Comment (mhegazy) [inline types/koa-route/index.d.ts]: insread of:
```ts
interface KoaRoutes { ... }
declare const routes: KoaRoutes;
export = routes;
```

```ts
 export var all: CreateRouteMethod;
 export var acl: RouteMethod;
 export var bind: RouteMethod;
 export var checkout: RouteMethod;
 export var connect: RouteMethod;
 export var copy: RouteMethod;
 export var delete: RouteMethod;
 export var del: RouteMethod;
 export var get: RouteMethod;
 export var head: RouteMethod;
 export var link: RouteMethod;
 export var lock: RouteMethod;
 export var msearch: RouteMethod;
 export var merge: RouteMethod;
 export var mkactivity: RouteMethod;
 export var mkcalendar: RouteMethod;
 export var mkcol: RouteMethod;
 export var move: RouteMethod;
 export var notify: RouteMethod;
 export var options: RouteMethod;
 export var patch: RouteMethod;
 export var post: RouteMethod;
 export var propfind: RouteMethod;
 export var proppatch: RouteMethod;
 export var purge: RouteMethod;
 export var put: RouteMethod;
 export var rebind: RouteMethod;
 export var report: RouteMethod;
 export var search: RouteMethod;
 export var subscribe: RouteMethod;
 export var trace: RouteMethod;
 export var unbind: RouteMethod;
 export var unlink: RouteMethod;
 export var unlock: RouteMethod;
 export var unsubscribe: RouteMethod;
```

**Review by mhegazy — CHANGES_REQUESTED** [review]

File: notNeededPackages.json [hunk 2]
```diff
@@ -355,6 +355,12 @@
             "asOfVersion": "1.1.2"
         },
         {
+            "libraryName": "path-to-regexp",
+            "typingsPackageName": "path-to-regexp",
+            "sourceRepoURL": "https://github.com/pillarjs/path-to-regexp",
+            "asOfVersion": "1.7.0"
```
Comment (andy-ms) [inline notNeededPackages.json]: Prefer `^1.7.0`.

File: types/koa-route/index.d.ts (hunk 1 above)
Comment (migstopheles) [inline types/koa-route/index.d.ts]: @mhegazy the reason for using the interface is that I get errors on the `delete` keyword due to strict mode. Figured it was better to keep strict mode and export in a different way, but I'm happy to make this change if you disagree, thanks

File: notNeededPackages.json [hunk 3]
```diff
@@ -355,6 +355,12 @@
             "asOfVersion": "1.1.2"
         },
         {
+            "libraryName": "path-to-regexp",
+            "typingsPackageName": "path-to-regexp",
+            "sourceRepoURL": "https://github.com/pillarjs/path-to-regexp",
+            "asOfVersion": "^1.7.0"
```
Comment (andy-ms) [inline notNeededPackages.json]: This should be an exact version `1.7.0`. It's not a dependency.

File: types/koa-route/package.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+    "dependencies": {
+        "path-to-regexp": "1.7.0"
```
Comment (andy-ms) [inline types/koa-route/package.json]: *This* should be `^1.7.0`.

Comment (migstopheles) [inline types/koa-route/package.json]: 💡 that makes much more sense. Updated, let me know if I should clean up the commit history for the branch.

Comment (andy-ms) [inline types/koa-route/package.json]: No need, we squash the commit history anyway.

**Pull request merged**